src/createNativeStackNavigator.js

```diff
@@ -234,6 +266,13 @@ class StackView extends React.Component {
       stackAnimation = 'none';
     }
 
+    const hasHeader =
+      this.props.navigationConfig?.headerMode !== 'none' &&
+      options.header !== null;
+
+    const isHeaderInModal = stackPresentation !== 'push' && hasHeader;
+    const isHeaderInPush = stackPresentation === 'push' && hasHeader;
```

Comment:
Like in `NativeStackView.tsx` I think `isHeaderInPush` variable is reduntant because when it's `InModal` it can't be `InPush` so just use `!isHeaderInModal` in line `302`.
```suggestion
    const isHeaderInModal = stackPresentation !== 'push' && hasHeader;
    
```

Reply:
Same as above.